commands/audit/scarunner.go

```diff
@@ -76,13 +76,14 @@ func buildDepTreeAndRunScaScan(auditParallelRunner *utils.SecurityParallelRunner
 		// Get the dependency tree for the technology in the working directory.
 		treeResult, bdtErr := buildDependencyTree(scan, auditParams)
 		if bdtErr != nil {
-			err = errors.Join(err, fmt.Errorf("audit command in '%s' failed:\n%s", scan.Target, bdtErr.Error()))
+			err = errors.Join(err, createErrorIfPartialResultsDisabled(auditParams, nil, fmt.Sprintf("Dependencies tree construction ha failed for the following target: %s", scan.Target), fmt.Errorf("audit command in '%s' failed:\n%s", scan.Target, bdtErr.Error())))
 			continue
 		}
 		// Create sca scan task
 		auditParallelRunner.ScaScansWg.Add(1)
 		_, taskErr := auditParallelRunner.Runner.AddTaskWithError(executeScaScanTask(auditParallelRunner, serverDetails, auditParams, scan, treeResult), func(err error) {
-			auditParallelRunner.AddErrorToChan(fmt.Errorf("audit command in '%s' failed:\n%s", scan.Target, err.Error()))
+			_ = createErrorIfPartialResultsDisabled(auditParams, auditParallelRunner, fmt.Sprintf("Failed to execute SCA scan for the following target: %s", scan.Target), fmt.Errorf("audit command in '%s' failed:\n%s", scan.Target, err.Error()))
```

Comment:
```suggestion
			err = createErrorIfPartialResultsDisabled(auditParams, auditParallelRunner, fmt.Sprintf("Failed to execute SCA scan for the following target: %s", scan.Target), fmt.Errorf("audit command in '%s' failed:\n%s", scan.Target, err.Error()))
```
error not handled, there is 2 times fmt with `scan.Target`, make sure error looks readable

Reply:
This is not really twice. createErrorIfPartialResultsDisabled works like this:
if allowPartialResults == true we log ONLY the message in the third argument (fmt.Sprintf("Failed to execute SCA scan for the following target: %s", scan.Target))
if allowPartialResults == false we use ONLY the error provided (fmt.Errorf("audit command in '%s' failed:\n%s", scan.Target, err.Error()) ). The error I provided is the exact same one we used to add to the error chain so there is not change in behaviour if allowPartialResults==false

Reply:
as for the unhandled error - this is suppose to be this way with a parallel runner is provided and there is nothing to do with the returned error